elastic/kibana
# Kibana discovery service

Resolves: #187696

This PR introduces Kibana Discovery Service for the TaskManager plugin.

- Creates a new SO type in the TaskManagerIndex.
- The SO has 2 fields: `id` (holds the kibana node id) and `last_seen` (timestamp of the last update applied by the node)
- Discovery Service in TM creates an SO on start and updates its last_seen field every 10s
- The service also deletes the SOs that haven't been updated in the last 5m, by checking the index every 1m.
- TM deletes its SO on plugin stop.
- Discovery Service provides an API (`getActiveKibanaNodes`) to get the active kibana nodes (last_seen field has been updated in the last 30s)

## To verify:
Run your Kibana locally and check the `.kibana_task_manager` index with the below query, there should be an SO and its last_seen field should be updated every 10s.

```
{
    "query": {
      "term": {
        "type": "background-task-node"        
      }
    },
    "size" : 10
}
```
---

The PR has been deployed to cloud as well, you can check the SOs for multiple Kibana instances there.
